### ISSUE TITLE
Linux arm64 support

### ISSUE THREAD

**Issue by jwillikers** [tom-james-watson/breaktimer-app#54]
I have a Pinebook Pro laptop which has arm64 processors. Would it be possible to release binaries for Linux arm64?

**Comment by tom-james-watson** [comment]
I'm not going to be able to do that at this point. I imagine you can build the app yourself. I'd also accept pull requests to get this working!

**Comment by jwillikers** [comment]
When I have time, I'll see if I can build it 👍

**Comment by tom-james-watson** [comment]
See https://github.com/tom-james-watson/breaktimer-app/blob/master/DEVELOPMENT.md 🙂

**Comment by jwillikers** [comment]
@tom-james-watson Thanks for the link.

I was able to get everything working and running in the dev environment.

On the default Manjaro KDE distribution on the Pinebook Pro, I installed yarn and npm.
```sh
$ pacman -S yarn npm
```

Then, I followed the instructions on your link, but I had to run the upgrade command to get around an error.
```sh
$ yarn
$ yarn upgrade
$ yarn install
$ yarn dev
```
While I was able to build the package, a fatal error occurred when running `yarn start`.
```sh
$ yarn build
$ yarn start
```
The output of the failed `yarn start` command is [here](https://paste.ubuntu.com/p/bwRKQmSNTn/)
The icon `node_modules/electron/dist/resources/app/resources/tray/icon.png` does not exist.

**Comment by tom-james-watson** [comment]
That's actually a problem on `master` that I hadn't noticed.

Note - I've since switched back to npm.

Are you saying the app works ok in dev mode (`npm run dev`)?

If you want to test the production version, you want to package the app up: `yarn package-linux`. Then you should be able to find the built version of the production app under the `release` folder.

**Comment by jwillikers** [comment]
The app works great in dev mode (`npm run dev`).

Running `npm run package` produces an AppImage which also works great!

**Comment by tom-james-watson** [comment]
That's great. I don't really have any way of building and publishing this myself at the moment, so I think this will just have to be unofficial support for now.

**Comment by jwillikers** [comment]
@tom-james-watson Yeah, that makes sense. I really appreciate the help and support! 🙏

Emulating [Arm64 on QEMU](https://wiki.debian.org/Arm64Qemu) and building it on that *might* not be too difficult. I can always test that out on one of my amd64 machines and let you know how it goes, if you think that's viable.

**Comment by tom-james-watson** [comment]
Yeah sure if it's something that is easily reproducible then it could be something we could look at adding.

**Comment by marijnbent** [comment]
Will this also work for the Apple M1 processor?

**Comment by tom-james-watson** [comment]
I haven't got the hardware to try this out, so I'm not going to be able to produce official builds any time soon.

As far as I can tell from https://github.com/electron-userland/electron-builder/issues/5392, if we upgrade electron and electron-builder then you should be able to manually build ARM64 macOS binaries without any changes.

**Comment by Catsvilles** [comment]
Latest version seems to be working fine on my Mac mini M1

**Comment by a-dows** [comment]
+1 for arm64 (M1 Mac) support, down to only two apps I run that are missing it!

**Comment by scristobal** [comment]
> +1 for arm64 (M1 Mac) support

+1 here too, got an M1 Pro recently and trying to avoid Rosseta2.

I have been a user of this app for a long time, so I would really appreciate to have it working natively on M1. 

Let me know if you are open to external contributions to fix this, or I could just test it out.

**Comment by tom-james-watson** [comment]
FYI as of 1.3.0 the macOS build now has native support for arm. You will need to manually download the update though, unfortunately. 

Still no support for linux arm though, I'm afraid.
